Update index.md (#294)
w3c/wai-curricula: content/2-6.md

#### Learning Outcomes for Topic

Students should be able to:

* explain how assistive technologies, such as text-to-speech and voice interaction, process roles to allow people with disabilities to identify and interact with widgets, and how missing role definitions make custom widgets unusable
* identify the appropriate roles for widgets, including buttons, carousels, sliders, tabs, treeviews, and others defined in the WAI-ARIA authoring practices, section [Design patterns and widgets](https://www.w3.org/TR/wai-aria-practices/#aria_ex)
* identify when HTML controls, such as `button`, `input`, and `option` as well as HTML links, can be used as the basis for custom widgets, such as buttons that are styled as switches, text input fields that are styled as sliders, and such

> * explain the accessibility benfits of using native HTML interactive elements, such as controls and links, as the basis for creating custom widgets rather than using non-interactive elements, such as `div`, `span`, and `section`, as the basis

Reviewer: ```suggestion
* explain the accessibility benefits of using native HTML interactive elements, such as controls and links, as the basis for creating custom widgets rather than using non-interactive elements, such as `div`, `span`, and `section`, as the basis
```

* code custom widgets using native HTML interactive elements, such as controls and links, with the necessary styling
* code custom widgets using additional WAI-ARIA attributes when native HTML interactive elements are not sufficient
* recite related requirements for content authors and designers to ensure that custom widgets are easy to understand

#### Teaching Ideas for Topic

Optional ideas to teach the learning outcomes:

* Show examples of standard and custom widgets such as buttons, checkboxes, and lists. Emphasize that widgets created with standard HTML elements that convey semantic information inherit roles by default. However, custom widgets need to have these roles expressed explicitly, especially when using HTML elements with not enough semantic information.

> * Compare use of standard and custom widgets with assistive technologies. Explain that accessibility support for WAI-ARIA properties may vary, so checking its current accessibility support is recommended. Mention that WAI-ARIA roles override the implicit roles of native HTML elements where it is used. For an explanation of how WAI-ARIA can provide both powerful and dangerous experiences see the WAI-ARIA Authoring Practices [Read me first](https://www.w3.org/TR/wai-aria-practices-1.1/#no_aria_better_bad_aria).

Reviewer: ```suggestion
* Compare use of standard and custom widgets with assistive technologies. Explain that accessibility support for WAI-ARIA properties may vary, so checking its current accessibility support is recommended. Mention that WAI-ARIA roles override the implicit roles of native HTML elements where they are used. For an explanation of how WAI-ARIA can provide both powerful and dangerous experiences see the WAI-ARIA Authoring Practices [Read me first](https://www.w3.org/TR/wai-aria-practices-1.1/#no_aria_better_bad_aria).
```

Fixing typo that I already had.

* Practical &mdash; Students are presented with a widget that reveals content based on user action and are asked to code it using `aria-haspopup` when necessary. Assess students' knowledge of the WAI-ARIA attribute `aria-haspopup`.
* Practical &mdash; Students are presented with custom widgets that have different states based on user action and are asked to code them accessibly. Assess students' capacity to code different states and its corresponding values.
* Practical &mdash; Students are presented with a widget that allows for drag-and-drop functionality and are asked to code it using WAI-ARIA attributes that convey such functionality. Assess students' knowledge of WAI-ARIA drag-and-drop attributes.

> * Practical &mdash; Students are presented with a chat application and are asked to code it appropriately so that it can be understood by people using text-to-speech. Assess students' knowledge of the types of live regions as well as its roles and properties.

Reviewer: ```suggestion
* Practical &mdash; Students are presented with a chat widget and are asked to code it appropriately so that it can be understood by people using text-to-speech. Assess students' knowledge of the types of live regions as well as its roles and properties.
```

Having mentions to a chat application here may be confusing, especially now that we have a dedicated module for rich applications. What about mentioning a chat widget?

Author: Good catch.
